reconcile/openshift_limitranges.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import semver
+import sys
+
+import utils.gql as gql
+import reconcile.openshift_base as ob
+
+from utils.openshift_resource import OpenshiftResource as OR
+from utils.defer import defer
+
+from reconcile.queries import NAMESPACES_QUERY
+
+QONTRACT_INTEGRATION = 'openshift-limitranges'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+SUPPORTED_LIMITRANGE_TYPES = (
+    'default',
+    'defaultRequest',
+    'max',
+    'maxLimitRequestRatio',
+    'min'
+)
+
+
+def construct_resources(namespaces):
+    for namespace in namespaces:
+        # Get the linked limitRanges schema settings
+        limitranges = namespace.get("limitRanges", {})
+
+        body = {
+            'apiVersion': 'v1',
+            'kind': 'LimitRange',
+            'metadata': {
+                'name': limitranges['name'],
+            },
+            'spec': {
+                'limits': []
+            }
+        }
+
+        # Build each limit item ignoring null ones
+        speclimit = {}
+        for l in limitranges['limits']:
+            for ltype in SUPPORTED_LIMITRANGE_TYPES:
+                if ltype in l and l[ltype] is not None:
+                    speclimit[ltype] = l[ltype]
+            body['spec']['limits'].append(speclimit)
+
+        resource = OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION)
+
+        # Create the resources and append them to the namespace
+        namespace["resources"] = [resource]
+
+    return namespaces
+
+
+def add_desired_state(namespaces, ri):
+    for namespace in namespaces:
+        for resource in namespace["resources"]:
+            ri.add_desired(
+                namespace['cluster']['name'],
+                namespace['name'],
+                resource.kind,
+                resource.name,
+                resource
+            )
+
+
+def set_delete_state(namespaces, ri):
```

Comment:
is this function intended to allow us to "take over" all existing LimitRanges and delete ones that we do not manage?
if so, i absolutely love it. i must admit that it was very hard to understand it though.

i would say that this pattern will be very reusable in the future, and even right now. for example, when we write `openshift-resourcequotas` (right?) and even for `openshift-network-policies` to not allow manual NetworkPolicy resources in a namespace.

it absolutely doesn't have to be in this PR, but let me know if you think its a good idea!

Reply:
The intent here is indeed to take over all LimitRanges and only apply ours. I'll admit this is a bit hack-ish and the preferable solution would probably be to support this in `ResourceInventory` with a function such as `ResourceInventory.delete(cluster, namespace, kind, name)`

I feel that was out of scope here and there is urgency in getting this integration deployed by next week

Reply:
sgtm!